persistence/trees/module.go

```diff
@@ -24,35 +31,52 @@ func (*treeStore) Create(bus modules.Bus, options ...modules.TreeStoreOption) (m
 	return m, nil
 }
 
-func Create(bus modules.Bus, options ...modules.TreeStoreOption) (modules.TreeStoreModule, error) {
-	return new(treeStore).Create(bus, options...)
+func Create(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, error) {
+	return new(TreeStore).Create(bus, options...)
 }
 
 // WithLogger assigns a logger for the tree store
-func WithLogger(logger *modules.Logger) modules.TreeStoreOption {
-	return func(m modules.TreeStoreModule) {
-		if mod, ok := m.(*treeStore); ok {
+func WithLogger(logger *modules.Logger) modules.ModuleOption {
+	return func(m modules.InitializableModule) {
+		if mod, ok := m.(*TreeStore); ok {
 			mod.logger = logger
 		}
 	}
 }
 
 // WithTreeStoreDirectory assigns the path where the tree store
 // saves its data.
-func WithTreeStoreDirectory(path string) modules.TreeStoreOption {
-	return func(m modules.TreeStoreModule) {
-		if mod, ok := m.(*treeStore); ok {
-			mod.treeStoreDir = path
+func WithTreeStoreDirectory(path string) modules.ModuleOption {
+	return func(m modules.InitializableModule) {
+		mod, ok := m.(*TreeStore)
+		if ok {
+			mod.TreeStoreDir = path
+		}
+	}
+}
+
+// WithTxIndexer assigns a TxIndexer for use during operation.
+func WithTxIndexer(txi indexer.TxIndexer) modules.ModuleOption {
+	return func(m modules.InitializableModule) {
+		mod, ok := m.(*TreeStore)
+		if ok {
+			mod.TXI = txi
 		}
 	}
 }
 
-func (t *treeStore) setupTrees() error {
-	if t.treeStoreDir == ":memory:" {
+func (t *TreeStore) GetModuleName() string  { return modules.TreeStoreModuleName }
+func (t *TreeStore) Start() error           { return nil }
```

Comment:
Is there a reason this is no longer a submodule when the Start and Stop functions are noop?

Reply:
I think this is necessary because the submodule stuff isn't merged into main yet and I'm not rebased on top of that branch.

Reply:
You could base off of `ibc/initial_stores_olsh..._delete_later` which includes the `ibc/initial_stores` and the submodule PRs if needed am using it as interim until merged in main

Reply:
I have rebased this on top of #896 which covers the Submodule updates.